src/lib/decorators.ts

```diff
@@ -224,6 +224,14 @@ export function query(selector: string) {
       enumerable: true,
       configurable: true,
     };
+    if (cache) {
+      const key = typeof name === 'symbol' ? Symbol() : `__${name}`;
+      descriptor.get = function(this: LitElement) {
+        return (this as unknown as {[key: string]: Element|null})[key as string] ||
```

Comment:
This will re-query every time if the query returns `null`... I suppose that's probably okay, but should it be documented behavior?

Reply:
Changed behavior to always cache first result.